hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/DataEntity.java

```diff
@@ -47,19 +47,22 @@ private DataEntity() {
         this.value = null;
     }
 
-    public DataEntity(CanonicalPath owner, DataRole role, StructuredData value, String identityHash) {
-        super(null, owner.extend(DataEntity.SEGMENT_TYPE, role.name()).get(), identityHash);
+    public DataEntity(CanonicalPath owner, DataRole role, StructuredData value, String identityHash, String contentHash,
+                      String syncHash) {
+        super(null, owner.extend(DataEntity.SEGMENT_TYPE, role.name()).get(), identityHash, contentHash, syncHash);
         this.value = value;
     }
 
     public DataEntity(CanonicalPath owner, DataRole role, StructuredData value, String identityHash,
-                      Map<String, Object> properties) {
-        super(owner.extend(DataEntity.SEGMENT_TYPE, role.name()).get(), identityHash, properties);
+                      String contentHash, String syncHash, Map<String, Object> properties) {
+        super(owner.extend(DataEntity.SEGMENT_TYPE, role.name()).get(), identityHash, contentHash, syncHash, properties);
         this.value = value;
     }
 
-    public DataEntity(CanonicalPath path, StructuredData value, String identityHash, Map<String, Object> properties) {
-        this(path.up(), DataRole.valueOf(path.getSegment().getElementId()), value, identityHash, properties);
+    public DataEntity(CanonicalPath path, StructuredData value, String identityHash, String contentHash,
+                      String syncHash, Map<String, Object> properties) {
+        this(path.up(), DataRole.valueOf(path.getSegment().getElementId()), value, identityHash, contentHash, syncHash,
+                properties);
```

Comment:
it looks, it's quite common that those `identityHash, contentHash, syncHash` are often used as params together, perhaps introducing some hashBundle object would simplify the method signatures (?)